pkg/cmd/admin/policy/reconcile_clusterroles.go

```diff
@@ -148,9 +155,60 @@ func (o *reconcileClusterOptions) ChangedClusterRoles() ([]*authorizationapi.Clu
 		}
 
 		if !kapi.Semantic.DeepEqual(expectedClusterRole.Rules, actualClusterRole.Rules) {
+			if o.Union {
+				modified := []authorizationapi.PolicyRule{}
+
+				// Find rules that don't exist exactly already
+				for _, modifiedRule := range actualClusterRole.Rules {
```

Comment:
Why not use this method first https://github.com/openshift/origin/blob/master/pkg/authorization/rulevalidation/policy_comparator.go#L11 before finding the rules to add?